examples/wordCount/lines2words/lines2words.cpp

```diff
@@ -99,52 +59,65 @@ std::vector<uint32_t> createConsumerPartitions(iceflow::Edge upstreamEdge,
   return consumerPartitions;
 }
 
-int main(int argc, const char *argv[]) {
-
-  if (argc != 3) {
-    std::string command = argv[0];
-    std::cout << "usage: " << command
-              << " <application-dag-file> <instance-number>" << std::endl;
-    return 1;
-  }
+void run(const std::string &nodeName, const std::string &dagFileName,
+         uint32_t consumerIndex) {
+  WordSplitter wordSplitter;
+  ndn::Face face;
 
-  std::string nodeName = "lines2words";
-  std::string dagFileName = argv[1];
-  int consumerIndex = std::stoi(argv[2]);
   auto dagParser = iceflow::DAGParser::parseFromFile(dagFileName);
-
-  std::string syncPrefix = "/" + dagParser.getApplicationName();
-  auto nodePrefix = generateNodePrefix();
   auto node = dagParser.findNodeByName(nodeName);
-  auto upstreamEdges = dagParser.findUpstreamEdges(node);
 
-  // TODO: Do this dynamically for all edges
-  auto downstreamEdge = node.downstream.at(0);
-  auto downstreamEdgeName = downstreamEdge.id;
-  auto numberOfProducerPartitions = downstreamEdge.maxPartitions;
+  auto upstreamEdge = dagParser.findUpstreamEdges(node).at(0).second;
+  auto upstreamEdgeName = upstreamEdge.id;
+  auto downstreamEdgeName = node.downstream.at(0).id;
+
+  auto consumerPartitions =
+      createConsumerPartitions(upstreamEdge.maxPartitions, consumerIndex);
 
   auto applicationConfiguration = node.applicationConfiguration;
   auto saveThreshold =
       applicationConfiguration.at("measurementsSaveThreshold").get<uint64_t>();
 
-  auto upstreamEdge = upstreamEdges.at(0).second;
-  auto upstreamEdgeName = upstreamEdge.id;
-  auto consumerPartitions =
-      createConsumerPartitions(upstreamEdge, consumerIndex);
+  auto iceflow = std::make_shared<iceflow::IceFlow>(dagParser, nodeName, face);
 
   ::signal(SIGINT, signalCallbackHandler);
-  measurementHandler =
-      new iceflow::Measurement(nodeName, nodePrefix, saveThreshold, "A");
+  measurementHandler = new iceflow::Measurement(
+      nodeName, iceflow->getNodePrefix(), saveThreshold, "A");
```

Comment:
I think ideally, it should be possible eventually to pass the DAG and the node name to the `Measurement` class instead of this "manual" configuration.

Reply:
I would also address this aspect via a follow-up PR since it is not related to the main IceFlow API.

Reply:
I just created #116 to keep track of this and will resolve this thread.